### PR TITLE
GG-21108 Implemented special isolated discovery.

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/web-console/web-console-server/src/main/resources/ignite-config.xml [hunk 1]
```diff
@@ -22,82 +22,15 @@
...
-                <property name="keyStoreFilePath" value="keystore/server.jks"/>
-                <property name="keyStorePassword" value="123456"/>
-                <property name="trustStoreFilePath" value="keystore/trust.jks"/>
-                <property name="trustStorePassword" value="123456"/>
-                <property name="setProtocol" value="SSL"/>
-              </bean>
-            </property>
-        -->
-
-        <property name="transactionConfiguration">
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/resources/ignite-config.xml]: Why we remove `transactionConfiguration`?

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    /** {@inheritDoc} */
+    @Override public Map<String, Object> attributes() {
+        return attrs;
+    }
+
+    /**
+     * @param attrs Node attributes.
+     */
+    public void attributes(Map<String, Object> attrs) {
+        this.attrs = attrs;
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: attrs should be unmodifiable, look at org.apache.ignite.spi.discovery.tcp.internal.TcpDiscoveryNode#setAttributes

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import org.apache.ignite.internal.ClusterMetricsSnapshot;
+import org.apache.ignite.internal.managers.discovery.IgniteClusterNode;
+import org.apache.ignite.lang.IgniteProductVersion;
+
+import static org.apache.ignite.internal.IgniteNodeAttributes.ATTR_NODE_CONSISTENT_ID;
+
+/**
+ * Special isolated node.
+ */
+public class IsolatedNode implements IgniteClusterNode {
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: May be will be better extend TcpDiscoveryNode, to reduce duplication of code?

File: modules/web-console/web-console-server/src/main/resources/ignite-config.xml (hunk 1 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/resources/ignite-config.xml]: Fixed

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java (hunk 3 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: I investigated this moment. And it seems that TcpDiscoveryNode not be easily sub-classed.

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpiSelfTest.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        c2.put(1, 2);
+
+        assertEquals(2, c2.get(1));
+    }
+
+    /**
+     * Test that isolated node restarts correctly.
+     */
+    @Test
+    public void testRestatrt() {
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpiSelfTest.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java (hunk 2 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: fixed

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    private Object consistentId;
+
+    /** Node attributes. */
+    private Map<String, Object> attrs;
+
+    /** */
+    private ClusterMetrics metrics = new ClusterMetricsSnapshot();
+
+    /** */
+    private Map<Integer, CacheMetrics> cacheMetrics = new HashMap<>();
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]:
```suggestion
    private volatile Map<Integer, CacheMetrics> cacheMetrics = Collections.<Integer, CacheMetrics>emptyMap();
```

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override public Map<Integer, CacheMetrics> cacheMetrics() {
+        return cacheMetrics;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setCacheMetrics(Map<Integer, CacheMetrics> cacheMetrics) {
+        this.cacheMetrics = cacheMetrics;
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]:
```suggestion
        this.cacheMetrics = cacheMetrics != null ? cacheMetrics : Collections.<Integer, CacheMetrics>emptyMap();
```

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    private final IgniteProductVersion ver;
+
+    /** Consistent ID. */
+    private Object consistentId;
+
+    /** Node attributes. */
+    private Map<String, Object> attrs;
+
+    /** */
+    private ClusterMetrics metrics = new ClusterMetricsSnapshot();
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]:
```suggestion
    private volatile ClusterMetrics metrics = new ClusterMetricsSnapshot();
```

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/discovery/IsolatedDiscoverySpi.java [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override public ClusterNode getNode(UUID nodeId) {
+        return locNode.id().equals(nodeId) ? locNode : null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean pingNode(UUID nodeId) {
+        return false;
```
Comment (nva) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/discovery/IsolatedDiscoverySpi.java]:
```suggestion
        return locNode.id().equals(nodeId);
```

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java (hunk 7 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: Fixed.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/discovery/IsolatedDiscoverySpi.java (hunk 8 above)
Comment (akuznetsov-gridgain) [reply modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/discovery/IsolatedDiscoverySpi.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java (hunk 6 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java (hunk 5 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java]: Fixed

**Pull request closed without merging**